qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -100,14 +117,26 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
 				QuteErrorCode errorCode = QuteErrorCode.getErrorCode(diagnostic.getCode());
 				if (errorCode != null) {
 					switch (errorCode) {
+					case UnknownProperty:
+						// CodeAction(s) to replace text with similar suggestions

```

Comment:
Why you are using data from diagnostic. It seems that you store the part name but codeaction can retrieve this information easily because codeaction can get the node and the part.